lib/index.js

```diff
@@ -44,29 +59,43 @@ function unCache (cache) {
  * context is the object (2nd argument) passed in to the reply.view method!
  */
 function compile (template, compileOpts) {
-  var baseOpts = Object.assign(DEFAULTS, compileOpts); // are we using this?
+  var baseOpts = Object.assign(DEFAULTS, compileOpts);
 
   return function render (context, renderOpts) {
+    var layoutFilePath, layoutFileName, content, View, compiledFileRoute;
     var mergedOpts = Object.assign(baseOpts, renderOpts);
-    var CACHE = loadTags(mergedOpts.filename);
-    var View = require(mergedOpts.filename); //eslint-disable-line
-    var output = mergedOpts.doctype;
+    var riotTag = template.split('>')[0].slice(1);
+
+    loadTags(mergedOpts.filename);
+    compiledFileRoute = mergedOpts.compiledFileRoute;
+    View = require(mergedOpts.filename); //eslint-disable-line
+    content = Riot.render(View, context)
+    + '\n<script src="https://cdn.jsdelivr.net/riot/2.6/riot.min.js" ></script>'
+    + '\n<script src="' + compiledFileRoute + '"></script>'
+    + '\n<script>\nriot.mount("'
+    + riotTag + '",' + JSON.stringify(context) + ')\n</script>'
+    ;
+
+    if (mergedOpts.layout) {
+      layoutFileName = mergedOpts.layout === true
+        ? 'layout.html'
+        : mergedOpts.layout
+      ;
+      layoutFilePath = Path.join(mergedOpts.layoutPath, layoutFileName);
+
+      content = loadLayout(layoutFilePath).replace('<<<RIOT>>>', content);
+    }
 
-    output += Riot.render(View, context);
     /*
-     * Delete the view and layout Tags from the require cache
+     * Delete the view and layout html from the require cache
      * so we don't need to restart the app to see view changes.
      * Skipped By default when NODE_ENV=production`.
      */
     if (mergedOpts.removeCache) {
-      // Do we need a layout option?
-      // if (mergedOpts.layout) {
-      //   delete require.cache[require.resolve(layoutPath)];
-      // }
-      unCache(CACHE);
+      unCache();
     }
 
-    return output;
+    return '<!DOCTYPE html>' + content;
```

Comment:
previously we allowed the `'<!DOCTYPE html>'` to be _overridden_? but my reading of MDN: https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5/Introduction_to_HTML5#Declaring_that_the_document_contains_HTML5_mark-up_with_the_HTML5_doctype is that this is not _necessary_ because HTML5 always has the simplified Doctype.